hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1186,6 +1266,73 @@ func (r *NodePoolReconciler) getConfig(ctx context.Context,
 	return strings.Join(allConfigPlainText, "\n---\n"), missingConfigs, utilerrors.NewAggregate(errors)
 }
 
+func (r *NodePoolReconciler) getTunedConfig(ctx context.Context,
+	nodePool *hyperv1.NodePool,
+) (configsRaw string, err error) {
+	var configs []corev1.ConfigMap
+	var allConfigPlainText []string
+	var errors []error
+
+	for _, config := range nodePool.Spec.TunedConfig {
+		configConfigMap := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      config.Name,
+				Namespace: nodePool.Namespace,
+			},
+		}
+		if err := r.Get(ctx, client.ObjectKeyFromObject(configConfigMap), configConfigMap); err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		configs = append(configs, *configConfigMap)
+	}
+
+	for _, config := range configs {
+		manifestRaw := config.Data[tunedConfigKey]
+		manifest, err := defaultAndValidateTunedConfigManifest([]byte(manifestRaw))
+		if err != nil {
+			errors = append(errors, fmt.Errorf("configmap %q failed validation: %w", config.Name, err))
+			continue
+		}
+
+		allConfigPlainText = append(allConfigPlainText, string(manifest))
+	}
+
+	// These configs are the input to a hash func whose output is used as part of the name of the user-data secret,
```

Comment:
this comment does not makes sense here.

Reply:
Thanks for catching this. I reworded this comment to be more in line with the current logic.